great_expectations/datasource/fluent/interfaces.py

```diff
@@ -340,7 +341,7 @@ class Datasource(
     # data context method `data_context.sources.add_my_datasource` method.
 
     # class attrs
-    asset_types: ClassVar[List[Type[DataAsset]]] = []
+    asset_types: ClassVar[Sequence[Type[DataAsset]]] = []
```

Comment:
@NathanFarmer Quick question: going forward, when do we use `List`, `Set`, etc. versus `Sequence`?  Thanks.

Reply:
It depends on the use case. I was running into issues when trying to use two lists in `asset_types` because lists are invariant. I wanted to make this a `tuple`, but that would require updating every `Datasource` in all of `fluent`. Using the more general `Sequence` allows us to use `list` where it already exists and enables covariance for `asset_types`.